postal-injection/src/main/java/com/google/sps/servlets/ShipEngine.java

```
@@ -0,0 +1,96 @@
+package com.google.sps;
+
+import java.lang.RuntimeException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Scanner;
+
+/** 
+* Single instance allowed of a class that reads all CSV information into objects and stores them
+* in a hash map of ZipCode String to ZipCode object containing all order information.
+*/
+public class ShipEngine {
+  private static boolean initialized = false;
+  private static HashMap<String, ZipCodeData> zipcodeDataMap;
+  private static final String CSV_FILE_NAME = "src/main/java/com/google/sps/servlets/data/data.csv";
+  public ShipEngine() {
+    if (this.initialized) {
+      throw new RuntimeException("A ShipEngine has already been initialized.");
+    } else {
+      zipcodeDataMap = new HashMap<>();
+      
+      File file = new File(CSV_FILE_NAME);
+      try {
+        initializeZipCodeAndOrderFromFile(file);
+      } catch (FileNotFoundException e) {
+        System.out.println(e);
+      }
```

Instead of suppressing the exception maybe just let it propagate (by removing try-catch)? I think that would make it easier to debug?